react-main/src/app/components/Game/index.js

```diff
@@ -2,53 +2,67 @@ import React, { Component } from 'react';
 
 import Board from '../Board';
 
+import { connect } from 'react-redux';
+import { selectSquare, selectHistory, getWinner } from '../../../redux/game/actions';
+
 import styles from './styles.scss';
-import calculateWinner from './utils.js';
 
 class Game extends Component {
-  state = { history: [{ squares: Array(9).fill(null) }], xIsNext: true, stepNumber: 0 };
-
-  jumpTo = move => this.setState({ stepNumber: move, xIsNext: move % 2 === 0 });
-
-  handleClick = i => {
-    const history = this.state.history;
-    const squares = history[this.state.stepNumber].squares.slice();
-    if (calculateWinner(squares) || squares[i]) {
-      return;
-    }
-
-    squares[i] = this.state.xIsNext ? 'X' : 'O';
-    this.setState({
-      history: history.concat([{ squares }]),
-      xIsNext: !this.state.xIsNext,
-      stepNumber: this.state.stepNumber + 1
-    });
+
+  handleSquareClick = i => (
+      this.props.square(i)
+  );
+
+  handleHistoryClick = step => {
+      return this.props.historySelected(step);
+  };
+
+  componentDidMount(){
+     this.props.winner()
+  };
+
+  componentWillUpdate(){
+     this.props.winner()
   };
 
   render() {
-    const history = this.state.history;
-    const squares = history[this.state.stepNumber].squares;
-    const winner = calculateWinner(squares);
-    const moves = history.map((step, move) => {
+    const moves = this.props.history.map((step, move) => {
       const desc = move ? `Go to move #${move}` : 'Go to game start';
       return (
         <li key={move}>
-          <button onClick={() => this.jumpTo(move)}>{desc}</button>
+          <button onClick={() => this.handleHistoryClick(move)}>{desc}</button>
         </li>
       );
     });
-    const status = winner ? `Winner: ${winner}` : `Next player: ${this.state.xIsNext ? 'X' : 'O'}`;
     return (
       <div className={styles.game}>
         <div className={styles.gameInfo}>
```

Comment:
It looks like you don't need this div. You could have a container div inside Board (if you don't already have it) and pass the className as prop.

```

<div className={styles.gameInfo}>
<Board className={styles.gameInfo} squares={this.props.history[this.props.current].squares} onClick={this.handleSquareClick} />

// Board/index.js
render() {
  <div className={className}>
    ...
  </div>
}
```

Reply:
what about this?

Reply:
what about this?